pkg/signup/signup_client.go

```diff
@@ -0,0 +1,99 @@
+package signup
+
+import (
+	"fmt"
+	crtapi "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"strings"
+)
+
+type signupClient struct {
+	ClientSet *kubernetes.Clientset
+}
+
+func NewSignupClient() (*signupClient, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
```

Comment:
@sbryzak i think it just moved a little: https://github.com/codeready-toolchain/registration-service/pull/23/files#diff-9033005564475e0bb99f8773d4829417R31

i think this should be the job the client - unless you think otherwise? i was thinking the client could encapsulate all this specific code for getting a client. but if you want, we can remove the client and do it differently.

Reply:
@tinakurian I discussed the client with Alexey today and we decided on a slight refactor to bring it more inline with the kubernetes API.